pyccel/ast/core.py

```diff
@@ -1649,9 +1647,52 @@ def name(self):
         else:
             return self.func
 
+class FunctionCallArgument(Basic):
+    """
+    An argument passed in a function call
+
+    Parameters
+    ---------
+    value   : PyccelAstNode
+              The expression passed as an argument
+    keyword : str
+              If the argument is passed by keyword then this
+              is that keyword
+    """
+    __slots__ = ('_value', '_keyword')
+    _attribute_nodes = ('_value',)
+    def __init__(self, value, keyword = None):
+        self._value = value
+        self._keyword = keyword
+        super().__init__()
+
+    @property
+    def value(self):
+        """ The value passed as argument
+        """
+        return self._value
+
+    @property
+    def keyword(self):
+        """ The keyword used to pass the argument
+        """
+        return self._keyword
+
+    @property
+    def has_keyword(self):
+        """ Indicates whether the argument was passed by keyword
+        """
+        return self._keyword is not None
+
+    def __repr__(self):
+        if self.has_keyword:
+            return 'FunctionCallArgument({} = {})'.format(self.keyword, repr(self.value))
+        else:
+            return 'FunctionCallArgument({})'.format(repr(self.value))
+
 class Argument(PyccelAstNode):
```

Comment:
For the sake of clarity and in order to avoid any confusion in the future, I think that we should rename this class as `FunctionDefArgument`.

Reply:
Fixed 11c6d097

Reply:
52783e7